tests/test_archive_querier.py

```diff
@@ -238,8 +238,8 @@ def get_multiple_pages(query_function, query_args):
         page = query_function(*query_args, cursor=cursor)
         page_len = len(page)
         assert page_len <= MAX_RESULTS
-        if len(results) == 0:
-            assert page.cursor is not None
```

Comment:
We actually want to leave this condition as-is. It checks that the first page of the multi-page request has a cursor. I think you should add another test condition that looks like this:

```
if page.cursor is not None:
    assert page_len > 0
```

This says that a page without results can only be a final page with no cursor.